### PR TITLE
feat(dashboard): surface stale-code CC sessions (MCP-staleness Part B)

### PR DESCRIPTION
**Pull request by WingedGuardian**
## What

Long-lived Claude Code sessions keep running the MCP-tool code they imported at
startup. A deploy restarts genesis-server but never the CC-launched stdio MCP
subprocesses, so a session silently runs **pre-deploy code** until it is
restarted. Part A (already merged) hard-guards the one *dangerous* stale tool
(`procedure_store`). This is **Part B: passive visibility** — a dashboard badge
so the operator can *see* which sessions predate the last code-changing deploy
and restart them by choice.

Deliberately **not** an alert: on an actively-developed install a
"stale-session-exists" condition is true almost continuously (many code-changing
deploys/day × sessions that live for days), so an alarm would fire
near-permanently. This is a passive indicator, computed at one site, that never
blocks anything and fails open.

## How

- `enumerate_cc_slots()` gains a `started_at` field — the process wall-clock start
  (`btime + /proc/<pid>/stat` field 22 `/ SC_CLK_TCK`), a faithful proxy for a
  session's MCP code version (the MCP children spawn with the parent and never
  reload). Additive key; the other two callers are unaffected.
- New CRUD reader `last_code_changing_update()` returns the most recent deploy
  that actually changed code (`new_commit != old_commit`), so a no-op `update.sh`
  re-run never flags a session running identical code.
- The CC-sessions detail route flags each live proc that started before that
  deploy (`stale_code` / `stale_since` / `deploy_commit`). Fail-open: unknown
  start time, empty history, or an unparseable timestamp → not stale. Both matched
  sessions and unmatched live slots carry it, so there is no blind spot.
- An amber “stale · restart for `<sha>`” badge renders on the Sessions tab and the
  overview modal, with a tooltip explaining which deploy landed after the session
  started.

## Deploy path

Runtime code + templates in genesis-server (the always-fresh Flask process) →
lands on the next `git pull` + server restart (`update.sh`). No schema change, no
host paths, no new store, no autonomous behavior (a passive read → no settings
lever needed). Empty-state correct by construction: empty `update_history` →
`None` → nothing flagged.

## Verification

- New unit tests: `read_proc_start_iso` (parse edges incl. comm-with-parens,
  missing pid/btime, malformed stat), `last_code_changing_update` (skips no-op
  re-runs and non-success rows, empty → None), and the route (stale/not-stale
  boundary, empty-state, fail-open on unknown start time, unmatched-slot coverage).
- Targeted + full `test_observability/` and `test_dashboard/` dirs green; the
  other `enumerate_cc_slots` consumers (leak-alert, cap-detection) unaffected by
  the additive key.
- Real-path E2E against the live DB + `/proc`: correctly flags the genuinely-stale
  live sessions; fail-open paths confirmed.
- Reviewed by the Genesis code-reviewer (DONE, no findings ≥ threshold).

Follow-up: b61771db

### CODE REVIEW

**Review by WingedGuardian — COMMENTED** [review]
Structural review: CLEAN — no pattern violations found.
PII/Secrets/Wording scan: CLEAN

Reviewed `cc_sessions.py`, `cc_slots.py`, `update_history.py`, and all three test files. No ERRORs, WARNINGs, or NOTEs on either pass. Raw SQL in `update_history.py` is correctly placed in the CRUD module. All new public functions have docstrings. Deferred imports inside `cc_sessions_detail()` follow established Genesis pattern. Test coverage added for all new paths (`read_proc_start_iso`, `last_code_changing_update`, `_stale_against_deploy` boundary cases, unmatched-slot staleness).

---
_Generated by [Claude Code](https://claude.ai/code)_

File: src/genesis/dashboard/routes/cc_sessions.py [hunk 1]
```diff
@@ -44,19 +44,49 @@
...
+    """
+    if not deploy or not started_at:
+        return False, None, None
+    completed_at, new_commit = deploy
+    st = _parse_iso_utc(started_at)
+    ct = _parse_iso_utc(completed_at)
+    if st is None or ct is None:
+        return False, None, None
+    if st < ct:
+        return True, completed_at, (new_commit or None)
```
[the remark this answers is not in the document]
Comment (WingedGuardian) [reply src/genesis/dashboard/routes/cc_sessions.py]: Addressed in v2 (commit dd90ef02). The badge no longer uses timestamps — it now compares each session's captured **spawn commit** against the deployed commit via the shared `commit_identity.is_stale` predicate (identity **and** time), the exact verdict Part A's guard uses. A session started mid-deploy (same commit) or ahead of the last recorded deploy is no longer flagged; verified by unit tests + real-path E2E.
